### ISSUE TITLE
Warning `MaxListenersExceededWarning`

### ISSUE THREAD

**Issue by slinkydeveloper** [restatedev/sdk-typescript#115]
While looking at the restate-chaos test failures, i stumbled on this warning:

```
TMzMzM5EAQYDhBS-01891ab91bf67db4b46ad36b5bf39e50] [Verify] : Invoking function.
[restate] [2023-07-03T07:48:06.291Z] DEBUG: [interpreter.CommandInterpreter-FwoTCg10ZXN0LTI4MTMzMzM5EAQYDhBS-01891ab91bf67db4b46ad36b5bf39e50] [Verify] : Function completed successfully.
[restate] [2023-07-03T07:48:06.335Z] DEBUG: [verifier.CommandVerifier-EwoNdGVzdC0yODEzMzMzORAEGA4=-01891ab91be67f6594d54c9026719c83] [Verify] : Function completed successfully.
(node:2) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 64 close listeners added to [Transform]. Use emitter.setMaxListeners() to increase limit
```

Probably unrelated to my research, but still worth noting. I ran the latest [e2e-node-services:latest image](https://github.com/restatedev/e2e/blob/main/services/node-services/Dockerfile).

**Comment by slinkydeveloper** [comment]
Just stumbled on this again while playing with the https://github.com/technomage/restate-perf

I see this on the test service immediately after it sends the milion of background calls.

```
(node:21041) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 drain listeners added to [ServerHttp2Stream]. Use emitter.setMaxListeners() to increase limit
(Use `node-18 --trace-warnings ...` to show where the warning was created)
```

Node and kernel version:

```
!11791 ➜ node --version
v18.18.2
!11792 ➜ uname -a
Linux fedora.fritz.box 6.6.3-100.fc38.x86_64 #1 SMP PREEMPT_DYNAMIC Tue Nov 28 20:36:17 UTC 2023 x86_64 GNU/Linux
```

**Comment by jackkleeman** [comment]
if its on ServerHttp2Stream i reckon this is just inherent to nodes http2 libs if you have enough concurrent streams

**Comment by slinkydeveloper** [comment]
the problem looks like you set a callback on the connection.send for each message

**Comment by slinkydeveloper** [comment]
I haven't seen this anymore, so i'm gonna close it.
